src/Symfony/Component/VarDumper/Caster/ConstStub.php

```diff
@@ -30,4 +30,27 @@ public function __toString()
     {
         return (string) $this->value;
     }
+
+    /**
+     * Creates ConstStub from bitfield flag.
+     *
+     * @param int    $value  The value of bitfield flag
+     * @param string $prefix The prefix that filters flags
+     *
+     * @return self
+     */
+    public static function fromFlag($value, $prefix)
+    {
+        $constants = get_defined_constants();
+        foreach ($constants as $c => $v) {
+            if ('' !== $prefix && 0 !== strpos($c, $prefix) || ($value & $v) !== $v) {
+                unset($constants[$c]);
+            }
+        }
+
+        asort($constants);
+        $name = implode(' | ', array_flip($constants)) ?: 0;
```

Comment:
@maidmaid the VarDumper is about showing you what the dumped value is. Assuming the ideal case when dumping is wrong, as people will generally dump stuff to debug their broken code. So we MUST make sure that the displayed data in the dump is actually what they dump.
And code expecting a binary flag gets an integer. In some cases, it could already be forbidding to pass invalid values (and so this use case would not happen), but this may not be the case all the time. And we should account for that in the dumper